flash/image/classification/integrations/baal/data.py

```diff
@@ -0,0 +1,121 @@
+    return torch.zeros(len(dataset))
+
+
+class ActiveLearningDataModule(LightningDataModule):
+    @requires("baal")
+    def __init__(
+        self,
+        labelled: Optional[DataModule] = None,
+        heuristic: "AbstractHeuristic" = BALD(reduction=np.mean),
+        map_dataset_to_labelled: Optional[Callable] = dataset_to_non_labelled_tensor,
+    ):
+        """The `ActiveLearningDataModule` handles data manipulation for ActiveLearning.
+
+        Args:
+            labelled: DataModule containing labelled train data for research use-case.
+                The labelled data would be masked.
+            heuristic: Sorting algorithm used to rank samples on how likely they can help with model performance.
+            map_dataset_to_labelled: Function used to emulate masking on labelled dataset.
+        """
+        self.labelled = labelled
+        self.heuristic = heuristic
+        self.map_dataset_to_labelled = map_dataset_to_labelled
+        self._dataset: Optional[ActiveLearningDataset] = None
+
+        if not self.labelled:
+            raise MisconfigurationException("The labelled `datamodule` should be provided.")
+
+        if not self.labelled.num_classes:
+            raise MisconfigurationException("The labelled dataset should be labelled")
+
+        if self.labelled and (
+            self.labelled._val_ds is not None
+            or self.labelled._test_ds is not None
+            or self.labelled._predict_ds is not None
+        ):
+            raise MisconfigurationException("The labelled `datamodule` should have only train data.")
+
+        self._dataset = ActiveLearningDataset(
+            self.labelled._train_ds, labelled=self.map_dataset_to_labelled(self.labelled._train_ds)
+        )
+
+        # hack to enable training.
+        if not len(self._dataset):
+            self.label(indices=[0])
+
+    @property
+    def has_unlabelled_data(self) -> bool:
+        return self._dataset.n_unlabelled > 0
+
+    @property
+    def num_classes(self) -> Optional[int]:
+        return getattr(self.labelled, "num_classes", None) or getattr(self.unlabelled, "num_classes", None)
+
+    @property
+    def data_pipeline(self) -> "DataPipeline":
+        return self.labelled.data_pipeline
+
+    def train_dataloader(self) -> "DataLoader":
+        self.labelled._train_ds = self._dataset
+        return self.labelled.train_dataloader()
+
+    def predict_dataloader(self) -> "DataLoader":
+        self.labelled._train_ds = self._dataset.pool
+        return self.labelled.train_dataloader()
+
+    def label(self, predictions: Any = None, indices=None):
+        if predictions and indices:
+            raise MisconfigurationException(
+                "The `predictions` and `indices` are mutually exclusive, pass only of one them."
+            )
+        if predictions:
+            uncertainties = [self.heuristic.get_uncertainties(np.asarray(p)) for idx, p in enumerate(predictions)]
```

Comment:
Can we call `self.heuristic.get_uncertainties(predictions)` directly?

Reply:
Yes, I am going to move it directly within `InferenceMCDropoutTask`, so we reduce memory usage.